client/src/components/FollowersView/FollowersView.jsx

```diff
@@ -1,29 +1,53 @@
 import React from 'react'
+import { useState, useEffect } from 'react'
 import UserCard from 'components/UserCard/UserCard'
 import './FollowersView.css'
+import { getUserProfileById } from 'utils/users'
 
 function FollowersView({
   profile,
-  followers
+  followers,
+  setPopupIsOpen,
+  setUserPopupId
 }) {
-  const users = followers ? profile.followers : profile.following
-  let userCards;
+  const [followerSearch, setFollowerSearch] = useState('')
+  const [users, setUsers] = useState(null)
+  const [displayUsers, setDisplayUsers] = useState([])
 
-  if (users.length === 0) {
-    userCards = <p>No followers</p>
-  } else {
-    userCards = users.map((item, idx) => {
-      return <UserCard key={idx} userId={item.userId} />
-    })
-  }
+  const userArray = followers ? profile.followers : profile.following
+
+  useEffect(() => {
+    const getUserInfo = async () => {
+      let newArray = []
```

Comment:
I don't think I can assign this map to an array because it contains promises and I am awaiting the promises outside of the loop.